machine_test.go

```diff
@@ -781,6 +801,18 @@ func TestCaptureFifoToFile(t *testing.T) {
 	}
 	defer os.Remove(fifoPath)
 
```

Comment:
You've changed the behavior of this test.  I think it'd be good to have tests for both behaviors rather than picking one over the other.